### PR TITLE
In C++ partitioned consumer, do not append partition index in subscription name

### PR DESCRIPTION
**Pull request by merlimat**
### Motivation

As described in #827, the C++ partitioned consumer is appending the partition id to the subscription name, unlike the Java partitioned consumer. That breaks the assumptions made by many tools and APIs to associate the same subscription id across all the partitions (eg: unsubscribe a subscription from all partitions at once).

### Modifications

Make C++ client to use the same convention as Java client in this case.

### CODE REVIEW

**Comment by jai1** [comment]
LGTM, but will cause the ppl using the old client to lose messages on the incorrect subscription name.

**Comment by merlimat** [comment]
> LGTM, but will cause the ppl using the old client to lose messages on the incorrect subscription name.

That's correct, though the idea is to fix it before anyone starts using it

**Comment by jai1** [comment]
retest this please

File: pulsar-client-cpp/tests/BasicEndToEndTest.cc [hunk 1]
```diff
@@ -429,6 +429,30 @@ TEST(BasicEndToEndTest, testPartitionedProducerConsumer)
...
+
+    // call admin api to make it partitioned
+    std::string url = adminUrl + "admin/persistent/prop/unit/ns/testPartitionedProducerConsumerSubscriptionName/partitions";
+    int res = makePutRequest(url, "3");
+
+    LOG_INFO("res = "<<res);
+    ASSERT_FALSE(res != 204 && res != 409);
+
+    Consumer partitionedConsumer;
+    Result result = client.subscribe(topicName, "subscription-A", consumer);
```
Comment (massakam) [inline pulsar-client-cpp/tests/BasicEndToEndTest.cc]: Is the variable name `consumer` (not `partitionedConsumer`) correct?

**Comment by massakam** [comment]
retest this please

**Comment by massakam** [comment]
retest this please

**Comment by massakam** [comment]
retest this please

**Comment by massakam** [comment]
retest this please

**Comment by massakam** [comment]
retest this please

**Pull request merged**